### PR TITLE
fix compilation errors of dma_sdk with ohw compiler

### PR DESCRIPTION
**Pull request by LuigiGiuffrida98**
@JuanSapriza @JoseCalero can you please review this PR?

### CODE REVIEW

**Comment by JoseCalero** [comment]
it is OK, but the PR is still on draft mode, why? are we waiting to more changings?

**Comment by JoseCalero** [comment]
ok, i have been debugging a bit, so far this is the output:

`===================================

    TESTING SINGLE MODE   

===================================

TEST:
Data size: 16
Signed: 0
Source type size: 8-bit
Destination type size: 8-bit
Padding not valid
Dimensionality not valid
tran: 10021     Error!
load: 512       Error!
laun: 512       Error!
...`

As you can see, we have problems when validating the transaction (see [dma_validate_transaction](https://github.com/esl-epfl/x-heep/blob/ea26a6f3ceaf634db3619b15f85b4c817f2d3aeb/sw/device/lib/drivers/dma/dma.c#L327))...
Specifically for the Padding and the dimensionality... @JuanSapriza could you help us debugging this?

**Comment by LuigiGiuffrida98** [comment]
Fixed all the issues, some fields of the DMA structures where not properly initialized, now everything is set to 0.
example_dma is now working both with riscv32-unknown and riscv32-corev toolchains.

**Review by davideschiavone — CHANGES_REQUESTED** [review]

File: sw/applications/example_dma/main.c [hunk 1]
```diff
@@ -225,6 +231,8 @@ uint8_t dma_window_ratio_warning_threshold()
 int main(int argc, char *argv[])
 {
 
+    // asm volatile ("ebreak");
```
[the remark this answers is not in the document]
Comment (LuigiGiuffrida98) [reply sw/applications/example_dma/main.c]: Done

**Comment by JuanSapriza** [comment]
Hello @LuigiGiuffrida98 @JoseCalero Just getting up to speed with all this. 
Is anything needed from my side or is this fixed?

**Pull request merged**